more methods addition in DataClient

rahulKQL opened the pull request:
## What changes this PR contains
Added below delegate methods in IBigtableDataClient & it's implementations
 - sampleRowKeys
 - sampleRowKeysAsync
 - readRows
 - readRowsAsync
 - readFlatRowsList
 - readFlatRows
 - readFlatRowsAsync

***
This PR is only for review

rahulKQL committed:
more methods addition in DataClient

Renamed test cases for DataClient

removed unnecessary excep declaration

romlogic commented on bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableDataClient.java:
```diff
@@ -94,4 +100,65 @@ Row readModifyWriteRow(ReadModifyWriteRow readModifyWriteRow)
+  List<KeyOffset> sampleRowKeys(String tableId);
+
+  /**
+   * Sample row keys from a table, returning a Future that will complete when the sampling has
```
line 113: Sample row keys from a table, returning a Future that will complete when the sampling has completed. 
line 133: @return a {@link ListenableFuture} that will finish when
   * all reads have completed.
line 158: @return a {@link com.google.common.util.concurrent.ListenableFuture} that will finish when all reads have completed.

Looking at all these lines, I would suggest the following: 
line 113: Sample row keys from a table; returns a Future that ends when the sampling has completed. 

line 133: @return a {@link ListenableFuture} that ends when
   * all reads have completed.

line 158: @return a {@link com.google.common.util.concurrent.ListenableFuture} that ends when all reads have completed.

rahulKQL committed:
Rebase and updated KeyOffSet#getKey

Renamed test cases for DataClient

removed unnecessary excep declaration

Renamed test cases for DataClient

removed unnecessary excep declaration
